src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -706,11 +706,13 @@ private function resolveServices($value, $file, $isParameter = false)
                 }
             }
             if ('tagged' === $value->getTag()) {
-                if (!\is_string($argument) || !$argument) {
-                    throw new InvalidArgumentException(sprintf('"!tagged" tag only accepts non empty string in "%s".', $file));
+                if (\is_string($argument) && $argument) {
+                    return new TaggedIteratorArgument($argument);
+                } elseif (\is_array($argument) && isset($argument['name']) && $argument['name']) {
+                    return new TaggedIteratorArgument($argument['name'], $argument['index_by'] ?? null, $argument['default_index_method'] ?? null);
```

Comment:
to ease spotting typos, we should validate that there are no extra keys

Reply:
Done